Resources/public/js/views/fields/ez-country-view.js

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+YUI.add('ez-country-view', function (Y) {
+    "use strict";
+    /**
+     * Provides the Country field view
+     *
+     * @module ez-country-view
+     */
+    Y.namespace('eZ');
+
+    /**
+     * The country field view
+     *
+     * @namespace eZ
+     * @class CountryView
+     * @constructor
+     * @extends eZ.FieldView
+     */
+    Y.eZ.CountryView = Y.Base.create('countryView', Y.eZ.FieldView, [], {
+
+        /**
+         * Returns the value to be used in the template. If the value is not
+         * filled, it returns undefined otherwise an object with a `link` and a
+         * `text` entries.
+         *
+         * @method _getFieldValue
+         * @protected
+         * @return Object
+         */
+        _getFieldValue: function () {
+            var config = this.get('config'),
+                field = this.get('field'),
+                fieldValue = field.fieldValue,
+                isMultiple = this.get('fieldDefinition').fieldSettings.isMultiple,
+                res = [];
+
+            if (field && fieldValue) {
+                Y.Array.each(fieldValue, function (country) {
+                    if (config[country]) {
+                        if(isMultiple) {
+                            res.push(config[country].Name);
+                        } else {
+                            res = config[country].Name;
```

Comment:
I would threat this case outside of loop to decrease a bit the imbrication level. So basically, the code could be:
```js
if (field && fieldValue) {
    if (!isMuliple ) {
        return config[fieldValue[0]].Name;
    }
    Y.Array.each(fieldValue, function (country) {
        // building a res when isMultiple
    })
}
return res;
```